nfs/rpc/tcp.go

```diff
@@ -51,3 +62,11 @@ func (t *tcpTransport) Write(buf []byte) (int, error) {
 func (t *tcpTransport) Close() error {
 	return t.wc.Close()
 }
+
+func (t *tcpTransport) SetTimeout(d time.Duration) {
+	t.timeout = d
+	if d == 0 {
+		var zeroTime time.Time
+		t.wc.SetDeadline(zeroTime)
```

Comment:
Just curious, what is the expected behavior here? If set to 0, will the connection timeout immediately?

Reply:
No, zero means "no timeout at all", this is a documented behavior of SetDeadline(), see https://golang.org/pkg/net/, a comment above SetDeadline()

Reply:
Great! Thanks!

Reply:
Hey @fdawg4l :) I believe the behavior if set to `0` is that the connection will never timeout. This is dictated by the behavior of the `Conn` struct [here](https://golang.org/pkg/net/#Conn). It also must be updated after each successful read and write if it is intended to be used as an "idle" timeout, though that would be up to the caller.

Reply:
ah I should have refreshed the page.

Reply:
@matthewavery Yeah, that's why I'm adding SetTimeout() calls in recv()/Write() methods -- so the caller just needs to call SetTimeout() once and then the code will update the deadlines and Do The Right Thing (c)

Reply:
yes, I missed the write side of things in my first glance through. My apologies.